rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+                'pointsize': pointsize}
+
+    kwargs.update({key: val for key, val in defaults.items()
+                   if key not in kwargs})
+
+    pointsize = kwargs.pop('pointsize')
+    image_artist = ax.scatter(photon_events[:, 0], photon_events[:, 1],
+                              s=photon_events[:, 2]*pointsize, **kwargs)
+    return image_artist
+
+
+def plot_pcolor(ax, photon_events, cax=None, bins=None, **kwargs):
+    """Make an pcolorfast image of the photon_events
+    This is effcient when there are a large RIXS intensity
+
+    Parameters
+    ----------
+    ax : matplotlib axes object
+        axes for plotting on
+    photon_events : array
+        two column x, y, Iph photon locations and intensities
+    cax : matplotlib axes object
+        use to plot colorbar to a specific axis
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * None chooses bin edges from int/2, int/2 +1 etc.
+              with partially populated edges removed.
+            * If int, the number of bins for the two dimensions (ny=nx=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+
+    **kwargs : dictionary
+        passed onto matplotlib.pyplot.imshow.
+    Returns
+    ----------
+    image_artist : matplotlib artist object
+        artist from image plot
+    cb_artist : matplotlib artist object
+        artist from colorbar associated with plot
+    """
+    ax.cla()
+
+    image_output = photon_events_to_image(photon_events, bins=bins)
+    x_centers, y_centers, image, x_edges, y_edges = image_output
+
+    defaults = {'cmap': 'gray',
+                'vmin': np.nanpercentile(image, 1),
+                'vmax': np.nanpercentile(photon_events[:, 2], 99)}
+
+    kwargs.update({key: val for key, val in defaults.items()
+                   if key not in kwargs})
+
+    xlim = (x_edges.min(), x_edges.max())
+    ylim = (y_edges.min(), y_edges.max())
+    image_artist = ax.pcolorfast(xlim, ylim, image, **kwargs)
+    cb_artist = plt.colorbar(image_artist, ax=ax, cax=cax)
+    ax.axis('tight')
+
+    return image_artist, cb_artist
+
+
+def poly(x, x2=0., x1=0., x0=500.):
+    """Second order polynominal function for fitting curvature.
+    Returns x2*x**2 + x1*x + x0
+    """
+    return x2*x**2 + x1*x + x0
+
+
+def get_curvature_offsets(photon_events, bins=None):
+    """ Determine the offests that define the isoenergetic line.
+    This is determined as the maximum of the cross correlation function with
+    a reference taken from the center of the image.
```

Comment:
This doc string is not enough information for me to follow what the function does. Can we expand it a little?

Reply:
Will do